static/js/src/PurchaseModal/components/Taxes/Taxes.tsx

```diff
@@ -48,10 +57,36 @@ const Taxes = ({ product, quantity }: TaxesProps) => {
     quantity,
     VATNumber: values.VATNumber,
   });
+  const genericPurchaseMutation = useMakePurchase();
 
   const onSaveClick = () => {
     setIsEditing(false);
-    taxMutation.mutate();
+    if (isGuest) {
```

Comment:
`isGuest` is not enough to decide which option to go for. We must also check if there is an existing `accountId` for the user. We call `preview` only when user not `isGuest` and there is and account ID associated with the user logged in. I think the account ID gets stored in the window.